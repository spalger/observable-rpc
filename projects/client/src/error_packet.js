export const errorPacketToError = params => {
  const {
    statusCode = '?',
    error = 'Unknown Error',
    message = '',
    ...extraParams
  } = params

  return Object.assign(
    new Error(
      `RPC[${statusCode}:${error}]${
        message && message !== error ? ` - ${message}` : ''
      }`
    ),
    {
      statusCode,
      type: error,
      ...extraParams,
    }
  )
}
