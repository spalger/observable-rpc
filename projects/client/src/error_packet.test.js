import { errorPacketToError } from './error_packet'

describe('errorPacketToError', () => {
  it('converts Boom output to an error object as expected', () => {
    expect(
      errorPacketToError({
        statusCode: 401,
        error: 'Unauthorized',
        message: 'You must specify a username and password',
        missing: ['username', 'password'],
      })
    ).toMatchSnapshot()
  })
})
