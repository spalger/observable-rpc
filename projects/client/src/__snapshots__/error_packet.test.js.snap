// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errorPacketToError converts Boom output to an error object as expected 1`] = `[Error: RPC[401:Unauthorized] - You must specify a username and password]`;
